NuRadioReco/utilities/trace_utilities.py

```diff
@@ -98,18 +97,242 @@ def get_channel_voltage_from_efield(station, electric_field, channels, detector,
+        else:
+            RMSNoise = None
+
+        signal_energy_fluence = f_signal * dt * conversion_factor_integrated_signal
+
+        # calculate error if RMSNoise is known:
+        if RMSNoise is not None and return_uncertainty:
+            signal_window_duration = sum(signal_window_mask) * dt if signal_window_mask is not None else len(times) * dt
+            signal_energy_fluence_error = (4 * np.abs(signal_energy_fluence / conversion_factor_integrated_signal) * RMSNoise ** 2 * dt + 2 * signal_window_duration * RMSNoise ** 4 * dt) ** 0.5  * conversion_factor_integrated_signal
+        else:
+            signal_energy_fluence_error = np.zeros(3)
+    
+    elif method == "rice_disttribution":
+        signal_energy_fluence = np.zeros(len(electric_field_trace))
+        signal_energy_fluence_error = np.zeros(len(electric_field_trace))
+        for i_pol in range(len(electric_field_trace)):
+            noise_estimators, frequencies_window = get_noise_fluence_estimators(
+                trace = electric_field_trace[i_pol, :],
+                times = times,
+                signal_window_mask = signal_window_mask,
+                **estimator_kwargs
+                )
+            estimators, variances = get_signal_fluence_estimators(
+                trace = electric_field_trace[i_pol, :],
+                times = times,
+                signal_window_mask = signal_window_mask,
+                noise_estimators = noise_estimators,
+                **estimator_kwargs
+                )
+
+            #sample frequency (after the windowing) in MHz
+            delta_f = frequencies_window[1] - frequencies_window[0]
+
+            #to convert the amplitudes squared into energy fluence units
+            conversion_factor = conversion_factor_integrated_signal
+
+            #correcting for selecting positive frequencies
+            one_side_spectrum_corr_factor = 1 #np.sqrt(2)
+
+            #get the fluence of the trace summing up the frequency estimators and converting in eV/m^2
+            fluence_freq = np.sum(estimators) * ((dt * one_side_spectrum_corr_factor) **2) * delta_f * conversion_factor
+
+            #get the variance of the trace fluence summing up the frequency variances and converting in (eV/m^2)^2
+            fluence_freq_variance = np.sum(variances) * (((dt * one_side_spectrum_corr_factor) **2) * delta_f * conversion_factor) **2
+
+            #get the fluence uncertainty as the root square of the variance
+            fluence_freq_error = np.sqrt(fluence_freq_variance)
+
+            signal_energy_fluence[i_pol] = fluence_freq
+            signal_energy_fluence_error[i_pol] = fluence_freq_error
+
+    if return_uncertainty:
+        return signal_energy_fluence, signal_energy_fluence_error
+    else:
+        return signal_energy_fluence
+
+def get_noise_fluence_estimators(trace, times, signal_window_mask, spacing_noise_signal=20*units.ns, relative_taper_width=0.142857143, use_median_value=False):
```

Comment:
Is this and the following function `get_signal_fluence_estimators` meant to be used outside of `get_electric_field_energy_fluence`? If not you could consider making them private.